packages/date-picker/src/DatePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,223 @@
+  id,
+  buttonRef = useRef<HTMLButtonElement>(null),
+  description,
+  onChange,
+  labelText,
+  isDisabled = false,
+  classNameOverride,
+  disabledDates,
+  disabledDaysOfWeek,
+  disabledRange,
+  disabledBeforeAfter,
+  disabledBefore,
+  disabledAfter,
+  firstDayOfWeek = 1,
+  initialMonth,
+  selectedDateRange,
+  value,
+  ...inputProps
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [referenceElement, setReferenceElement] =
+    useState<HTMLDivElement | null>(null)
+  const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(
+    null
+  )
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const { styles, attributes } = usePopper(referenceElement, popperElement, {
+    modifiers: [
+      {
+        name: "offset",
+        options: {
+          offset: [0, 15],
+        },
+      },
+    ],
+    placement: "bottom-start",
+  })
+
+  const handleOpenClose = () => {
+    setIsOpen(!isOpen)
+  }
+
+  const handleReturnFocus = () => {
+    if (buttonRef.current) {
+      buttonRef.current.focus()
+    }
+  }
+
+  const disabledDays = calculateDisabledDays(
+    disabledDates,
+    disabledDaysOfWeek,
+    disabledRange,
+    disabledBeforeAfter,
+    disabledBefore,
+    disabledAfter
+  )
+
+  const handleDayClick = (day: Date, modifiers: DayModifiers) => {
+    /** react-day-picker will fire events for disabled days by default.
+     *  We're checking here if it includes the CSS Modules class for disabled
+     *  on the modifier to then return early.
+     * */
+    if (Object.keys(modifiers).includes(defaultCalendarClasses.disabled)) {
+      return
+    }
+
+    if (!selectedDateRange) return
+    onChange(DateUtils.addDayToRange(day, selectedDateRange))
+  }
+
+  const modifiers: RangeModifier = {
+    from: selectedDateRange?.from,
+    to: selectedDateRange?.to,
+  }
+
+  return (
+    <div ref={wrapperRef}>
+      <div ref={setReferenceElement}>
+        <Label disabled={isDisabled} htmlFor={id} labelText={labelText} />
+        <button
+          className={cx(
+            datePickerStyles.button,
+            datePickerStyles.withStartIconAdornment,
+            {
+              [datePickerStyles.disabled]: isDisabled,
+            }
+          )}
+          id={id}
+          disabled={isDisabled}
+          ref={buttonRef}
+          onClick={handleOpenClose}
+          {...inputProps}
+        >
+          <div className={datePickerStyles.startIconAdornment}>
+            <Icon icon={dateStart} role="presentation" />
+          </div>
+          <span className={datePickerStyles.value}>
+            {selectedDateRange?.from ? value : undefined}
+          </span>
+        </button>
+      </div>
+      {isOpen && (
+        <FocusOn
+          scrollLock={false}
+          onDeactivation={handleReturnFocus}
+          onClickOutside={() => {
+            handleOpenClose()
+          }}
+          onEscapeKey={() => {
+            handleOpenClose()
+          }}
```

Comment:
Hmm I can't seem to replicate the full screen drop out on Chrome. What browser are you in?

Reply:
@mcwinter07 realising you're in Firefox but trying to replicate that issue now and having trouble. What is the screen reader you're using? It seems to be adding the first cell but not the second which is strange.

Full disclosure, we may not be able to do the aria-label part easily, as we don't have access to the API to pass aria-labels in without a lot of work. This may be a v2 implementation as it's working on Chrome with their default screen reader extension.

As for the escaping -> exit to full page, that is strange so we'll try and get that working before this goes in.

Reply:
@mcwinter07 I'm using Firefox and can escape with the focus returning to the input. Can you send a video?

Reply:
@dougmacknz FYI in regards to the screenreader

Reply:
@lloydstubber using Macs native screen reader - so hopefully you'll be able to replicate that. I might also jump to my other OS to test it out in NVDA since I know that's one of the more popular open source readers for uses on windows machines

Reply:
@mcwinter07 FYI preventDefault doesn't fix the issue either

Reply:
😆 Well dang!